packages/ketcher-core/src/domain/entities/DrawingEntitiesManager.ts

```diff
@@ -684,23 +702,29 @@ export class DrawingEntitiesManager {
     }
   }
 
-  public deletePolymerBond(polymerBond: PolymerBond) {
+  public deletePolymerBond(polymerBond: PolymerBond | HydrogenBond) {
     const command = new Command();
+    const firstAttachmentPoint =
+      polymerBond.firstMonomer.getAttachmentPointByBond(
+        polymerBond,
+      ) as AttachmentPointName;
+    const secondAttachmentPoint =
+      polymerBond.secondMonomer?.getAttachmentPointByBond(
+        polymerBond,
+      ) as AttachmentPointName;
 
     const operation = new PolymerBondDeleteOperation(
       polymerBond,
       this.deletePolymerBondChangeModel.bind(this, polymerBond),
-      this.finishPolymerBondCreationModelChange.bind(
-        this,
-        polymerBond.firstMonomer,
-        polymerBond.secondMonomer as BaseMonomer,
-        polymerBond.firstMonomer.getAttachmentPointByBond(
-          polymerBond,
-        ) as AttachmentPointName,
-        polymerBond.secondMonomer?.getAttachmentPointByBond(
-          polymerBond,
-        ) as AttachmentPointName,
-      ),
+      (_polymerBond?: PolymerBond) =>
```

Comment:
It's a bit confusing, `polymerBond` type is specified as a union in the `deletePolymerBond` yet here it's just `PolymerBond` and I didn't see any type narrowing?